Description: properly handle Ubuntu's openssl having OP_NO_SSLv3 forced on by default
Author: Marc Deslauriers <marc.deslauriers@canonical.com>
Forwarded: yes, http://bugs.python.org/issue25724

Index: b/Lib/test/test_ssl.py
===================================================================
--- a/Lib/test/test_ssl.py
+++ b/Lib/test/test_ssl.py
@@ -752,7 +752,8 @@ class ContextTests(unittest.TestCase):
             self.assertEqual(ssl.OP_ALL | ssl.OP_NO_TLSv1 | ssl.OP_NO_SSLv3,
                              ctx.options)
             ctx.options = 0
-            self.assertEqual(0, ctx.options)
+            # Ubuntu has OP_NO_SSLv3 forced on by default
+            self.assertEqual(0, ctx.options & ~ssl.OP_NO_SSLv3)
         else:
             with self.assertRaises(ValueError):
                 ctx.options = 0
